### PR TITLE
#170400135 - Approve or Reject Bookings

### PR DESCRIPTION
**Pull request by Cheza-Dzabala**
#### What does this PR do?

- Adds accept and reject to booking actions
- Adds cases to bookings reducer
- Adds a pendingBookings.js file
- Adds pending bookings to routes
- Adds pending bookings to menu component
- Fixes responsiveness feedback on `/accommodations/:slug/` page
- Implements feedback on making all accommodations search page responsive
- Updates all accommodations UI

#### Description of Task to be completed?
 - A supplier must be able to log into the application and approve or reject bookings

#### How should this be manually tested?
- Clone this repo
- Run npm install & run npm run dev
- Log in as a supplier
- Navigate to bookings / pending
- Attempt to accept or reject a booking

#### Any background context you want to provide?
 N / A
#### Screenshots (if appropriate)
Screen Shot 2020-01-03 at 11 29 46
Screen Shot 2020-01-03 at 11 29 53
Screen Shot 2020-01-03 at 11 30 00
Screen Shot 2020-01-03 at 11 30 07

###### FEEDBACK
Screen Shot 2020-01-05 at 21 11 22
Screen Shot 2020-01-05 at 22 36 00

#### Questions:
N/A

### CODE REVIEW

**Review by ngireric123 — APPROVED** [review]
LGTM

**Comment by ngireric123** [comment]
bro kindly increase ur test coverage to 90

**Review by nignanthomas — APPROVED** [review]
LGTM @Cheza-Dzabala

File: src/views/bookings/PendingBookings.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+      isLoading: true,
+      userId: null,
+      operation: null,
+      isProcessing: false,
+    };
+  }
+
+  componentDidMount = async () => {
+    const { getPendingBookings } = this.props;
+    const userInfo = isAuthenticated();
```
Comment (nakiwuge) [inline src/views/bookings/PendingBookings.js:33]: Add a new line

File: src/views/bookings/PendingBookings.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    await getPendingBookings();
+    this.setState({
+      isLoading: false,
+    });
+  }
+
+  actionSwitch = async (action, id) => {
+    this.setState({
+      isProcessing: true,
+    });
```
Comment (nakiwuge) [inline src/views/bookings/PendingBookings.js:49]: Add a new line

File: src/views/bookings/PendingBookings.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        this.dispatchRejectBooking(id);
+    }
+  }
+
+  dispatchRejectBooking = async (id) => {
+    const { getPendingBookings } = this.props;
+    this.setState({
+      operation: 'rejected',
+    });
+    const { rejectBooking } = this.props;
```
Comment (nakiwuge) [inline src/views/bookings/PendingBookings.js:67]: Add a new line

File: src/views/bookings/PendingBookings.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    await getPendingBookings();
+    this.setState({
+      isProcessing: false,
+    });
+  }
+
+  showBookings = () => {
+    const { bookings, errorMessage, status } = this.props;
+    const { operation, isProcessing } = this.state;
+    return (
```
Comment (nakiwuge) [inline src/views/bookings/PendingBookings.js:101]: Add a new line here

File: src/views/bookings/PendingBookings.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    <Container fluid>
+      <Row className="center-items">
+        <h1>You have no pending bookings</h1>
+      </Row>
+    </Container>
+  )
+
+  render() {
+    const { isLoading, userId } = this.state;
+    const { bookings } = this.props;
```
Comment (nakiwuge) [inline src/views/bookings/PendingBookings.js]: Add a new line here

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@Cheza-Dzabala good job however I have noticed some things that I would like you to change.
- When a user approves/rejects a booking a blank page is returned until the user refreshes the page.
- The  confirmation modal is not responsive

File: src/views/bookings/PendingBookings.js (hunk 4 above)
Comment (Cheza-Dzabala) [reply src/views/bookings/PendingBookings.js:101]: Added line, not sure why it's not reflecting

File: src/views/bookings/PendingBookings.js (hunk 3 above)
Comment (Cheza-Dzabala) [reply src/views/bookings/PendingBookings.js:67]: Added line, not sure why it's not reflecting

File: src/views/bookings/PendingBookings.js (hunk 2 above)
Comment (Cheza-Dzabala) [reply src/views/bookings/PendingBookings.js:49]: Added line, not sure why it's not reflecting

File: src/views/bookings/PendingBookings.js (hunk 1 above)
Comment (Cheza-Dzabala) [reply src/views/bookings/PendingBookings.js:33]: Added line, not sure why it's not reflecting

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**